### PR TITLE
Lightning: reduce the complexity of precheck

### PR DESCRIPTION
**Pull request by okJiang**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/41895

Problem Summary:

### What is changed and how it works?

reduce the number of calling `GetAllTableStructures`, it is enough just one

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by dsdashun — APPROVED** [review]

File: br/pkg/lightning/restore/get_pre_info.go [hunk 1]
```diff
@@ -398,6 +398,7 @@ get_struct_from_src:
 			return nil, errors.Annotatef(err, "get create table statement from schema file error: %s", tableFileMeta.Name)
 		}
 		theTableInfo, err := newTableInfo(createTblSQL, 0)
+		log.L().Info("generate table info from SQL", zap.Error(err), zap.String("sql", createTblSQL), zap.String("table_name", tableFileMeta.Name), zap.String("db_name", dbSrcFileMeta.Name))
```
Comment (okJiang) [inline br/pkg/lightning/restore/get_pre_info.go:401]: Maybe remove later

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**